### DISCUSSION TITLE
Is there any limitation of SSG using Nextjs?

### DISCUSSION THREAD

**Discussion by mickmak-busicat** [vercel/vercel#8441] [Help] [answered]
Hi there,

I am trying to deploy my Nextjs web app to Vercel. One main feature of the web is serving many static products page with SSG and `fallback: true`. So while build time, it will not generate all product pages. Product pages are only generated when user browse a product. We want all the generated pages can be cached as long as possible until next build deployed and there are quite a lot of products in our database too.

So my question is about is there any limit for the cached page?
I found there is a max cache time limit of 31 days for Vercel's Edge Network cache.
https://vercel.com/docs/concepts/edge-network/caching#limits

Does it also apply to the cached page generated by SSG? If yes, can we override this setting?
It seems there is no mention about the maximum number of generated pages in the website for all the plan. Is there really no such limit in this case?

Any answer would help.

Mick

**Comment by amyegan** [comment, marked as answer]
That's right. The cache limit is 31 days. This applies to all static files and there is not a way to exceed this limit. Static files are automatically cached at the edge after the first request, and the cache will be refreshed on the next request after it expires.

> Static files are cached for up to 31 days. If a file is unchanged, it can persist across deployments, as their hash caches static files. However, the cache is effectively invalidated when you redeploy, so we always serve the latest version.

There is no limit on number of pages, so long as you don't exceed any other limits.

You can learn more here: 
- [Static File Caching](https://vercel.com/docs/concepts/edge-network/caching#static-files-caching)
- [Edge Network FAQs](https://vercel.com/docs/concepts/edge-network/frequently-asked-questions)
- [General Limits](https://vercel.com/docs/concepts/limits/overview#general-limits)
- [15,000 limit for source files, no limit on number of output files](https://vercel.com/docs/concepts/limits/overview#files)
- [Incremental Static Regeneration](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration#)
- [ISR, Not Just Caching](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration#not-just-caching)

**Reply by mickmak-busicat** [reply]
Thank you very much! I will check them all.
